server/actions_specs.go

```diff
@@ -198,6 +203,46 @@ func (s *RegistryServer) getApiSpecRevision(ctx context.Context, name names.Spec
 	return message, nil
 }
 
+// GUnzippedBytes uncompresses a slice of bytes.
+func GUnzippedBytes(input []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(input)
+	zr, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.ReadAll(zr)
+}
+
+// GetApiSpecContents handles the corresponding API request.
+func (s *RegistryServer) GetApiSpecContents(ctx context.Context, req *rpc.GetApiSpecContentsRequest) (*httpbody.HttpBody, error) {
+	var spec *rpc.ApiSpec
+	var err error
+	if name, err := names.ParseSpec(req.GetName()); err == nil {
+		spec, err = s.getApiSpec(ctx, name, rpc.View_FULL)
+	} else if name, err := names.ParseSpecRevision(req.GetName()); err == nil {
+		spec, err = s.getApiSpecRevision(ctx, name, rpc.View_FULL)
+	}
+	if err != nil {
+		return nil, err
+	}
+	if spec != nil {
+		contents := spec.Contents
+		contentType := spec.MimeType
+		if (strings.Contains(contentType, "+gzip")) {
+			contents, err = GUnzippedBytes(contents)
+			contentType = strings.Replace(contentType, "+gzip", "", 1)
```

Comment:
Do we want to replace +gzip even if it's not a suffix of the content type? If it exists multiple times, do we only want to remove the first instance of it? These both seem like very unusual and maybe invalid cases.

Consider using strings.HasSuffix instead of strings.Contains above, and strings.TrimSuffix here if covers everything we want. If not, maybe a comment would be useful to point out an example of an unusual case where this would function differently.

Reply:
I originally used HasSuffix and discovered that this broke when mime types had version parameters as I'd proposed in https://apigee.github.io/registry/media_types.html - for example `application/x.openapi+gzip;version=3.0.0`. I agree that more careful parsing would be good to consider/plan for the future.